web/documentation/events.md

```diff
@@ -0,0 +1,118 @@
+    |     |                    |           +->[ spfhistory ]
+    |     +-------------+------+                            
+    |                   |                                   
+    |                   |                                   
+    |                   +-------------------->[ spfrequest ]
+    |                   |                                   
+    |               send|request                            
+    |                   |                                   
+    |                   +--------------+                    
+    |                   |              |                    
+    |                   |       history|pushstate           
+    |                   |              |                    
+    |                   +--------------+                    
+    |                   |                                   
+    |                   |                                   
+    |            receive|response                           
+    |                   |                                   
+    |                   +-------------------->[ spfprocess ]
+    |                   |                                   
+    |                   |                                   
+    |            process|response                           
+    |                   |                                   
+    |                   +----------------------->[ spfdone ]
+    |                   |                                   
+    |                   |                                   
+    +-------------------+                                   
+                        |                                   
+                        |                                   
+                        v                                   
+
+
+## Event Descriptions
+
+All events in the [API][] are defined as [spf.Event][] objects.  A list of
+the events and their descriptions follows:
+
+**`spfclick`**  
+Fired when handling a click on a valid link (e.g. one with a valid `spf-link`
+class or parent element).  Use as an early indication that navigation will
+happen or to provide element-level UI feedback.
+
+**`spfhistory`**  
+Fired when handling a `popstate` history event, indicating the user has gone
+backward or forward; similar to `spfclick`.
+
+**`spfrequest`**  
+Fired before a request for navigation is sent.  Use to handle the beginning of
+navigation and provide global-level UI feedback (i.e. start a progress bar).
+This event is fired before a request is sent for all types of navigation:
+clicks, back/forward, and API calls.  (Note that this event is fired even if a
+response is fetched from cache and no actual network request is made.)
+
+**`spfprocess`**  
+Fired when a response has been received, either from the network or from
+cache, before it is processed.  Use to update UI feedback (i.e. advance a
+progress bar) and dispose event listeners before content is updated.
+
+**`spfdone`**  
```

Comment:
Yes, I was planning on putting the multipart events on the multipart page and the script/style events on the resources page, and then I was going to provide links from the bottom of this page to those.  WDYT?